openapi/index.yaml

```diff
@@ -1,49 +1,265 @@
+      - name: ordering
+        in: query
+        type: string
+        enum:
+          - distance
+          - alphabetic
+        required: true
+        description: Ordering type, by distance or alphabetical
+    get:
+      operationId: getOfflineMerchants
+      summary: List of merchants with a physical address
+      description: |
+        List of merchants with physical address, ordered by distance from the user by default
+      responses:
+        "200":
+          description: List of physical merchants for the given query parameters
+          schema:
+            "$ref": "#/definitions/OfflineMerchants"
+
+  /merchants/{merchantId}:
+    parameters:
+      - name: merchantId
         in: path
         type: string
         required: true
-        description: test input.
+        description: Merchant Identifier
     get:
-      operationId: echo
-      summary: Echo API
-      description: Some API description here.
+      operationId: getMerchant
+      summary: Retrieve merchant details
+      description: |
+        Retrieve merchant details together with published discounts
       responses:
         "200":
-          description: Found.
+          description: Found
           schema:
-            "$ref": "#/definitions/Input"
-          examples:
-            application/json:
-              id: "10000AAAA"
-              output: "hello world !"
-        "401":
-          description: Unauthorized
+            "$ref": "#/definitions/Merchant"
         "404":
-          description: Not found.
-      parameters: []
+          "$ref": "#/responses/NotFound"
+
+parameters:
+  Name:
+    name: merchantName
+    in: query
+    description: Text used for searching merchants by their name
+    required: false
+    type: string
+  ProductCategories:
+    name: productCategories
+    in: query
+    description: List of product categories to filter the results
+    required: false
+    collectionFormat: csv
+    type: array
+    items:
+      type: string
+
 definitions:
-  Input:
+  Merchant:
+    type: object
+    required:
+      - id
+      - name
+      - description
+      - discounts
+    properties:
+      id:
+        type: string
+      name:
+        type: string
+      description:
+        type: string
+      websiteUrl:
+        type: string
+      discounts:
+        type: array
+        items:
+          "$ref": "#/definitions/Discount"
+      addresses:
+        type: array
+        items:
+          "$ref": "#/definitions/Address"
+
+  Discount:
+    type: object
+    required:
+      - id
+      - name
+      - startDate
+      - endDate
+      - productCategories
+      - discount
+    properties:
+      id:
+        type: string
+      name:
+        type: string
+      description:
+        type: string
+      startDate:
+        type: string
+        format: date
+      endDate:
+        type: string
+        format: date
+      productCategories:
```

Comment:
I'm not sure but I'd rather check if `productCategories` items are enums :)

Reply:
now the product categories are defined as enum aligned with specs